numpy/random/setup.py

```diff
@@ -52,78 +54,87 @@ def generate_libraries(ext, build_dir):
     PCG64_DEFS = []
     # One can force emulated 128-bit arithmetic if one wants.
     #PCG64_DEFS += [('PCG_FORCE_EMULATED_128BIT_MATH', '1')]
+    depends = ['__init__.pxd', 'c_distributions.pxd', '_bit_generator.pxd']
+
+    # npyrandom - a library like npymath
+    npyrandom_sources = [
+        'src/distributions/logfactorial.c',
+        'src/distributions/distributions.c',
+        'src/distributions/random_mvhg_count.c',
+        'src/distributions/random_mvhg_marginals.c',
+        'src/distributions/random_hypergeometric.c',
+    ]
+    config.add_installed_library('npyrandom',
+        sources=npyrandom_sources,
+        install_dir='lib',
+        build_info={
+            'include_dirs' : [],  # empty list required for creating npyrandom.h
+            'extra_compiler_args' : (['/GL-'] if is_msvc else []),
+        })
 
     for gen in ['mt19937']:
         # gen.pyx, src/gen/gen.c, src/gen/gen-jump.c
-        config.add_extension('_{0}'.format(gen),
-                             sources=['_{0}.c'.format(gen),
-                                      'src/{0}/{0}.c'.format(gen),
-                                      'src/{0}/{0}-jump.c'.format(gen)],
+        config.add_extension(f'_{gen}',
+                             sources=[f'_{gen}.c',
+                                      f'src/{gen}/{gen}.c',
+                                      f'src/{gen}/{gen}-jump.c'],
```

Comment:
as per previous discussion, I am replacing formatting with f-strings since I am touching this file. Are there more here I missed?